<?xml version='1.0' ?>

<launch>

  <arg name="use_sim_time" default="true" description="Use the /clock topic for time to sync with simulation"/>
  <arg name="config_file" description="The config file that provides important parameters for setting up the adapter"/>
  <arg name="nav_graph_file" description="The graph that this fleet should use for navigation"/>
  <arg name="server_uri" default="" description="The URI of the api server to transmit state and task information."/>
  <arg name="output" default="screen"/>

  <!-- Fleet adapter -->
  <group if="$(var use_sim_time)">
    <node pkg="rmf_fleet_adapter"
          exec="easy_full_control"
          output="both">

      <param name="nav_graph_file" value="$(var nav_graph_file)"/>
      <param name="config_file" value="$(var config_file)"/>
      <param name="use_sim_time" value="$(var use_sim_time)"/>
      <param name="server_uri" value="$(var server_uri)"/>
    </node>
  </group>

  <group unless="$(var use_sim_time)">
    <node pkg="rmf_fleet_adapter"
          exec="easy_full_control"
          output="both">

      <param name="nav_graph_file" value="$(var nav_graph_file)"/>
      <param name="config_file" value="$(var config_file)"/>
      <param name="use_sim_time" value="$(var use_sim_time)"/>
      <param name="server_uri" value="$(var server_uri)"/>
    </node>
  </group>

</launch>